### PR TITLE
Feature/logwidg3

### PR DESCRIPTION
**Pull request by hmoco**

### CODE REVIEW

File: website/static/css/home.css [hunk 1]
```diff
@@ -0,0 +1,4 @@
+/*
```
Comment (samanehsan) [inline website/static/css/home.css]: Is this file necessary?

File: website/static/js/logTextParser.js [hunk 2]
```diff
@@ -110,11 +125,11 @@ var LogPieces = {
     contributors: {
         view: function (ctrl, logObject) {
             var contributors = logObject.embeds.contributors;
-            if(paramIsReturned(contributors, logObject)) {
+            /*if(paramIsReturned(contributors, logObject)) {
```
Comment (samanehsan) [inline website/static/js/logTextParser.js]: Did you mean to leave this commented out?

**Comment by samanehsan** [comment]
I moved a folder `A/` from Dropbox to OSF Storage, and the project log says:
"Taylor Swift moved A/ from Dropbox in Blank Space to A/ in OSF Storage in Blank Space" 

But the recent activity log says "Taylor Swift copied A/ to A/": 
screen shot 2016-02-18 at 4 15 19 pm

I also think giving information about the file provider would be useful, since the project logs include that info.

File: website/views.py [hunk 3]
```diff
@@ -124,7 +124,10 @@ def dashboard(auth):
 
 @must_be_logged_in
 def home(auth):
-    return {}
+    user = auth.user
+    return {
+        'userId': user._id,
```
Comment (samanehsan) [inline website/views.py]: `userId` is being passed to `home.mako`, but I don't see it being used anywhere. Can it be removed?

File: website/static/js/home.js [hunk 4]
```diff
@@ -0,0 +1,8 @@
+var $ = require('jquery');
```
Comment (samanehsan) [inline website/static/js/home.js]: This file is missing `'use strict';` at the top

Comment (samanehsan) [inline website/static/js/home.js]: This file is missing a `'use strict';` at the top

Comment (samanehsan) [inline website/static/js/home.js]: Also, should this file go in the `js/pages` directory since it's a webpack entry point?

File: website/static/js/recentActivityWidget.js [hunk 5]
```diff
@@ -0,0 +1,345 @@
...
+                if (!self.cache[self.page] && self.page < self.lastPage){
+                    self.page = self.page + 1;
+                    self.getLogs(false, false, true);
+                    self.page = self.page - 1;
+                }
+                return;
+            }
+            var query = {
+                'embed': ['nodes', 'user', 'linked_node', 'template_node'],
+                'page': ((self.page/2) | 0) + 1,
```
Comment (samanehsan) [inline website/static/js/recentActivityWidget.js:57]: This should be `||` instead of `|`. There are a few more places that are using `|` so please do a quick search on this file and fix those as well!

File: website/static/js/recentActivityWidget.js [hunk 6]
```diff
@@ -0,0 +1,345 @@
+var $ = require('jquery');
```
Comment (samanehsan) [inline website/static/js/recentActivityWidget.js:3]: This file is also missing `'use strict;` at the top

File: website/static/js/recentActivityWidget.js (hunk 5 above)
Comment (hmoco) [inline website/static/js/recentActivityWidget.js:57]: No, actually this is not an `or`, it is a bitwise or, to allow for division of integer rounded down:

```
((7/3) || 0)
 = 2.3333333333333335
((7/3) | 0)
 = 2
```

There this is used so that the requested page 3, for example, corresponds to index (`self.page`) 6 and 7.

Comment (hmoco) [inline website/static/js/recentActivityWidget.js:57]: Everywhere this is used is for that purpose, to avoid using `Math.floor()`

File: api/users/permissions.py [hunk 7]
```diff
@@ -14,6 +14,13 @@ def has_object_permission(self, request, view, obj):
         else:
             return obj == request_user
 
+class CurrentUser(permissions.BasePermission):
+    """ Check to see if the request is coming from the currently logged user
+    """
+
+    def has_object_permission(self, request, view, obj):
+        assert isinstance(obj, User), 'obj must be a User, got {}'.format(obj)
+        return obj == request.user
```
Comment (samanehsan) [inline api/users/permissions.py:23]: Probably more reliable to compare by `_id` here

File: website/static/js/recentActivityWidget.js (hunk 5 above)
Comment (samanehsan) [inline website/static/js/recentActivityWidget.js:57]: Oh, okay. Thanks for the clarification!

File: api_tests/users/views/test_user_node_logs.py [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+
+        assert_equal(res.json['links']['meta']['total'], 2)
+
+    def test_no_auth(self):
+        res = self.app.get(
+            '/{0}users/{1}/logs/'.format(API_BASE, self.user._id),
+            expect_errors=True
+        )
+
+        assert_equal(res.status_code, 401)
```
Comment (samanehsan) [inline api_tests/users/views/test_user_node_logs.py:90]: For tests that give you 400-type errors, it's good to assert that the error message is what you expect it to be, just in case the test errors for the wrong reason.

**Comment by samanehsan** [comment]
Pass complete @hmoco! :octocat: I just had a few minor comments & questions.

File: api/users/permissions.py (hunk 7 above)
Comment (hmoco) [inline api/users/permissions.py:23]: I think I need to keep check objects, as chekcing `_id`s causes this to pop up `'AnonymousUser' object has no attribute '_id'`, any idea why?

File: api_tests/users/views/test_user_node_logs.py (hunk 8 above)
Comment (hmoco) [inline api_tests/users/views/test_user_node_logs.py:90]: No custom error messages are being passed, so I'm not sure another way to make sure the right error is being passed.

**Comment by hmoco** [comment]
@samanehsan, have responded, back to ya!

File: website/static/js/pages/home.js [hunk 9]
```diff
@@ -0,0 +1,10 @@
+'use strict';
+
+var $ = require('jquery');
+var m = require('mithril');
+var LogWrap = require('js/recentActivityWidget');
+require('css/home.css');
```
Comment (samanehsan) [inline website/static/js/pages/home.js]: Please remove this require statement, since `home.css` was deleted

File: website/static/js/logTextParser.js [hunk 10]
```diff
@@ -223,7 +224,7 @@ var LogPieces = {
         view: function (ctrl, logObject) {
             var destination = logObject.attributes.params.destination;
             if(paramIsReturned(destination, logObject)){
-                return m('a', {href: destination.url}, destination.materialized);
+                return m('span', [m('a', {href: destination.url}), destination.materialized, ' in ', source.addon]);
```
Comment (samanehsan) [inline website/static/js/logTextParser.js]: I'm getting a console error pointing to this line -- `Uncaught ReferenceError: source is not defined`

Comment (hmoco) [inline website/static/js/logTextParser.js]: oops. That should be `destination.addon`

Comment (samanehsan) [inline website/static/js/logTextParser.js]: Also, not sure if this is related, but project logs seem to only link the "destination" file:
screen shot 2016-02-22 at 10 51 09 am

But recentActivity logs link the "source" file:
screen shot 2016-02-22 at 10 51 24 am

Also, these project logs include the name of the project. I think it would be useful if the recent activity logs included that info as well. Especially because it's aggregating logs from multiple projects.

**Comment by samanehsan** [comment]
Also, it looks like "move" actions are still getting logged as "copy" actions.

**Comment by samanehsan** [comment]
Some of my logs are blank:
screen shot 2016-02-22 at 11 41 08 am

And the console error message is `There is no text entry in dictionary for the action :comment_restored`

**Comment by samanehsan** [comment]
The project logs for file comments are slightly different:
screen shot 2016-02-22 at 11 47 14 am

For consistency, I think that information should also be included in the recent activity logs. Right now both file comment and project comment logs look like this:  screen shot 2016-02-22 at 11 48 24 am

File: api/users/permissions.py (hunk 7 above)
Comment (samanehsan) [inline api/users/permissions.py:23]: `AnonymousUser` objects do not have ids, so you could do something like: 

```
if user.is_anonymous():
     return False
return obj.user._id == user._id
```

**Comment by samanehsan** [comment]
Pass complete @hmoco! :whale:

**Comment by hmoco** [comment]
@samanehsan back to you, would you mind testing the logs that were giving you problems again?

File: website/static/js/pages/home.js (hunk 9 above)
Comment (samanehsan) [inline website/static/js/pages/home.js]: @hmoco, pls remove `require('css/home.css');`

**Comment by samanehsan** [comment]
The blank logs are now rendering properly!

**Comment by samanehsan** [comment]
The file comment logs are still the same though -- is it possible to include which file was commented?

**Comment by samanehsan** [comment]
Back to you @hmoco! :octopus:

**Comment by samanehsan** [comment]
Oh, one more thing -- folders should not be linked because they will always go to a Not Found error page:  
screen shot 2016-02-22 at 2 18 33 pm

**Comment by hmoco** [comment]
The reason for the comment logs appearing that way is that it seems that the file params are not being serialized and sent as part of v2 log responses.

**Comment by hmoco** [comment]
@samanehsan have responded to the folder comment, I do not believe comment log serialization is within the scope of this PR, but if you think otherwise I can try and add that in.

**Comment by hmoco** [comment]
@samanehsan, here is the ticket for the improvement necessary to include the file links in the comment logs: https://openscience.atlassian.net/browse/OSF-5766

**Comment by samanehsan** [comment]
Looks good to me @hmoco! :umbrella:
